lab_5_scrapper/scrapper.py

```diff
@@ -127,11 +234,39 @@ def _extract_url(self, article_bs: BeautifulSoup) -> str:
         Returns:
             str: Url from HTML
         """
+        container_div = article_bs.find('div', class_='col-lg-9 col-md-12')
+        for link in container_div.find_all('a', href=True):
+            if link.get('href').startswith('/news/202'):
+                href = 'https://www.gazetavechorka.ru' + link.get('href')
+                if href and href not in self.urls:
+                    return href
 
     def find_articles(self) -> None:
         """
         Find articles.
         """
+        seed_urls = self.get_search_urls()
+        necessary_len = self.config.get_num_articles()
+
+        while len(self.urls) != necessary_len:
+            for url in seed_urls:
+                response = make_request(url, self.config)
+
+                if response.ok:
```

Comment:
rework to have code less nested (skip iteration if not OK)

Reply:
done